src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,115 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private int initialCapacity = 16;
+    private T[] table;
+    private int size;
+
+    @SuppressWarnings("unchecked")
+    public ArrayList() {
+        this.table = (T[]) new Object[initialCapacity];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == initialCapacity) {
+            increaseArray();
+        }
+        table[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("index " + index + "non exist");
+        } else if (size == initialCapacity) {
+            increaseArray();
+        }
+        T[] table;
+        int subSize = size;
+        table = this.table;
+        System.arraycopy(table, index, table, index + 1, subSize - index);
+        size = subSize + 1;
+        table[index] = value;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index <= size - 1 && index >= 0) {
```

Comment:
it should be separate method 
```suggestion
        checkIndex(index);
        return (T) table[index];
```